internal/state/graph/gateway.go

```diff
@@ -89,8 +95,38 @@ func buildGateway(gw *v1beta1.Gateway, secretMemoryMgr secrets.SecretDiskMemoryM
 		return nil
 	}
 
+	conds := validateGateway(gw, gc)
+
+	if len(conds) > 0 {
+		return &Gateway{
+			Source:     gw,
+			Valid:      false,
+			Conditions: conds,
+		}
+	}
+
 	return &Gateway{
 		Source:    gw,
-		Listeners: buildListeners(gw, secretMemoryMgr, gc),
+		Listeners: buildListeners(gw, secretMemoryMgr),
+		Valid:     true,
+	}
+}
+
+func validateGateway(gw *v1beta1.Gateway, gc *GatewayClass) []conditions.Condition {
+	var conds []conditions.Condition
+
+	if gc == nil {
+		conds = append(conds, conditions.NewGatewayInvalid("GatewayClass doesn't exist"))
```

Comment:
because the condition here and in line 120 and also the condition for addresses are Accepted False conditions, would it make sense to short circuit in this function and return just a single Condition?

Reply:
We can return a single condition here, but I was trying to stick with the pattern of returning a list of conditions and checking whether the list is greater than 0. 

It's also possible that the gateway class is invalid/missing _and_ an address is provided, so I was trying to allow that case. I know we just de-dupe in `statuses.go`, but the pattern established so far is to include all conditions when we build the graph and de-dupe in `statuses.go`.